### ISSUE TITLE
Users Specified `device` Variable Not Works as Expected when Scaling Down OSD

### ISSUE THREAD

**Issue by digglife** [ceph/ceph-ansible#5723]
**Bug Report**

What happened:

1. I deployed a ceph cluster, one of the nodes has 5 OSDs.

    ```yaml
    dedicated_devices:
    - /dev/sdb
    devices:
    - /dev/sdc
    - /dev/sdd
    - /dev/sde
    - /dev/sdf
    - /dev/sdg
    ```

2. I scaled the OSDs on this node down to 3. Delete 2 OSDs with `ceph osd rm`, then changed the relevant `host_var` to the following one.

    ```yaml
    dedicated_devices:
    - /dev/sdb
    devices:
    - /dev/sdc
    - /dev/sdd
    - /dev/sde
    ```
3. Redeployed the cluster. `systemd` services were still created for the OSDs of `sdf` and `sdg` thus the containers were still initialized.

What you expected to happen:

Only 3 OSDs created.

How to reproduce it (minimal and precise):

Just like above mentioned steps.

Share your group_vars files, inventory and **full** ceph-ansibe log

I can't share the full log. But I think I've found the root cause of this issue.

`ceph osd rm` just deletes the OSDs, not the lvm associated with them,
so the OSD ids collected by [`start_osds.yml`](https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-osd/tasks/start_osds.yml) will not changed at all, all the OSDs created before will be started.

```yaml
- name: systemd start osd
  systemd:
    name: ceph-osd@{{ item }}
    state: started
    enabled: yes
    masked: no
    daemon_reload: yes
  with_items: "{{ ((ceph_osd_ids.stdout | default('{}') | from_json).keys() | list) + osd_ids_non_container.stdout_lines | default([]) }}"
```

**Environment**:
* OS (e.g. from /etc/os-release): `RHEL 8.1`
* Kernel (e.g. `uname -a`): `4.18.0-147.24.2.el8_1.x86_64`
* Docker version if applicable (e.g. `docker version`): `podman version 1.6.4`
* Ansible version (e.g. `ansible-playbook --version`): `2.8.12`
* ceph-ansible version (e.g. `git head or tag or stable branch`): `4.0.25.2`
* Ceph version (e.g. `ceph -v`): `14.2.4-125.el8cp`

**Comment by dsavineau** [comment]
That's not a ceph-ansible problem but an issue on your side.

If you want to remove manually an OSD then you need to remove everything:

- stop and disable the systemd service
- ceph osd rm/purge/down/out
- zap the device/lv/vg/pv with ceph-volume.

All of this can be achieved automatically by running the shrink-osd.yml playbook [1]

Note that 4.0.25.2 has a regression on that playbook [2] which has been fixed in 4.0.26

[1] https://github.com/ceph/ceph-ansible/blob/master/infrastructure-playbooks/shrink-osd.yml
[2] https://bugzilla.redhat.com/show_bug.cgi?id=1854973

**Comment by digglife** [comment]
@dsavineau 

Thank you for the detailed explanation. I zapped the relevant devices and deleted the data dir manually. Now everything works fine. It's also great to know there is a playbook exactly for this.
